cmake/utils.cmake

```diff
@@ -256,52 +256,72 @@ endfunction()
 
 #
 # mayaUsd_copyDirectory( <target>
+#                        [SOURCE <directory>]
 #                        [DESTINATION <destination>]
-#                        [DIRECTORY <directory>])
+#                        [EXCLUDE <exclude_files>]
+#                        [GLOB <pattern>]
 #
+#   SOURCE        - directory to be copied.
 #   DESTINATION   - destination where directory will be copied into.
-#   DIRECTORY     - directory to be copied.
+#   EXCLUDE       - exclude files from copy.
+#   GLOB          - glob files with pattern (e.g *.cpp, *.py)
 #
 function(mayaUsd_copyDirectory target)
     cmake_parse_arguments(PREFIX
-        ""             # options
-        "DESTINATION"  # one_value keywords
-        "DIRECTORY"    # multi_value keywords
+        ""
+        "SOURCE;DESTINATION" # one value-keyword
+        "EXCLUDE;GLOB"       # multi_value keywords
         ${ARGN}
     )
 
-    if(PREFIX_DESTINATION)
-        set(destination ${PREFIX_DESTINATION})
+    # source
+    if(NOT PREFIX_SOURCE)
+        set(PREFIX_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}")
     endif()
+    get_filename_component(PREFIX_SOURCE "${PREFIX_SOURCE}" ABSOLUTE)
+
+    # destination
     if(NOT PREFIX_DESTINATION)
-        message(FATAL_ERROR "DESTINATION keyword is not specified.")
+        set(PREFIX_DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
     endif()
+    get_filename_component(PREFIX_DESTINATION "${PREFIX_DESTINATION}" ABSOLUTE)
 
-     if(PREFIX_DIRECTORY)
-        set(directory ${PREFIX_DIRECTORY})
-        get_filename_component(directory "${directory}" ABSOLUTE)
-        get_filename_component(dir_name "${directory}" NAME)
-    endif()
-    if(NOT PREFIX_DIRECTORY)
-        message(FATAL_ERROR "DIRECTORY keyword is not specified.")
+    # exclude
+    if(NOT PREFIX_EXCLUDE)
+        unset(PREFIX_EXCLUDE)
```

Comment:
Hey @ppt-adsk  exactly, I mainly added it to avoid UB when PREFIX_EXCLUDE is not specified, same thing for exclude_files and glob_files.  
This is based on what I was reading the other day:
https://gitlab.kitware.com/cmake/cmake/commit/546a32897839f015440ea0cc6c8f5b486359d26f